lib/THC/generic/THCTensorMath.cu

```diff
@@ -391,4 +391,42 @@ accreal THCTensor_(trace)(THCState *state, THCTensor *src_) {
   return trace;
 }
 
+#if defined(THC_REAL_IS_FLOAT) || defined(THC_REAL_IS_DOUBLE) || defined(THC_REAL_IS_HALF)
+
+void THCTensor_(linspace)(THCState *state, THCTensor *r_, real a, real b, long n) {
+  THCAssertSameGPU(THCTensor_(checkGPU)(state, 1, r_));
```

Comment:
and then in these functions, you pass the base address of `r_` to the operator which is kept as a member, and you either assert that `r_` is contiguous, or you call `newContiguous` to enforce it.